src/header-navigation/styled-components.js

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+import {styled, asPrimaryExport} from '../styles';
+import {ALIGN} from './index';
+
+export const Root = styled('nav', props => {
+  const {$theme} = props;
+  const {
+    sizing: {scale500},
+    typography: {font400},
+    colors: {border},
+  } = $theme;
+  return {
+    ...font400,
+    cursor: 'pointer',
+    display: 'flex',
+    paddingBottom: scale500,
+    paddingTop: scale500,
+    borderBottom: `1px solid ${border}`,
+  };
+});
+
+export const NavigationItem = styled('div', props => {
+  const {$theme} = props;
+  const {
+    sizing: {scale800},
+  } = $theme;
+  return {
+    alignSelf: 'center',
+    paddingLeft: scale800,
+  };
+});
+
+export const NavigationList = asPrimaryExport(
+  styled('div', props => {
+    const {$align, $theme} = props;
+    const {
+      sizing: {scale800},
+    } = $theme;
+    return {
+      display: 'flex',
+      ':first-child': {
+        padding: 0,
+      },
+      ':last-child': {
+        padding: 0,
+      },
+      flex: $align === ALIGN.right || $align === ALIGN.left ? 'none' : '1',
+      paddingLeft: scale800,
+      paddingRight: scale800,
+      justifySelf: $align,
+      justifyContent: $align,
+    };
+  }),
+  ['align'],
+);
```

Comment:
It might be nice to also flowtype this component since it's a primary export. 

Flow, might complain that this is incompatible with return type of asPrimaryExport, but you could always cast through any to fix that.

```js
export const NavigationList: React.ComponentType<
  {align: string, children: React.Node}
> = (asPrimaryExport( ... ): any);
```

Reply:
fixed. I wonder why flow didn't complain about it